packages/block-editor/src/components/block-fallback-web-version/index.native.js

```diff
@@ -0,0 +1,207 @@
+	icon,
+	title,
+	description,
+	actionButtonLabel,
+	customActions = EMPTY_ARRAY,
+} ) => {
+	const [ sendFallbackMessage, setSendFallbackMessage ] = useState( false );
+	const [ sendButtonPressMessage, setSendButtonPressMessage ] =
+		useState( false );
+
+	const {
+		blockName,
+		blockContent,
+		isUnsupportedBlockEditorSupported,
+		canEnableUnsupportedBlockEditor,
+		isEditableInUnsupportedBlockEditor,
+	} = useSelect(
+		( select ) => {
+			const { getBlock, getSettings } = select( blockEditorStore );
+			const { capabilities } = getSettings();
+
+			const block = getBlock( clientId );
+			const blockAttributes = block?.attributes || {};
+
+			const blockDetails = {
+				blockName: block?.name,
+				blockContent: serialize( block ? [ block ] : [] ),
+			};
+
+			// If the block is unsupported, use the `original` attributes to identify the block's name.
+			if ( blockDetails.blockName === 'core/missing' ) {
+				blockDetails.blockName = blockAttributes.originalName;
+				blockDetails.blockContent =
+					blockDetails.blockName === 'core/freeform'
+						? blockAttributes.content
+						: block?.originalContent;
+			}
+
+			return {
+				isUnsupportedBlockEditorSupported:
+					capabilities?.unsupportedBlockEditor === true,
+				canEnableUnsupportedBlockEditor:
+					capabilities?.canEnableUnsupportedBlockEditor === true,
+				isEditableInUnsupportedBlockEditor:
+					! UBE_INCOMPATIBLE_BLOCKS.includes(
+						blockDetails.blockName
+					),
+				...blockDetails,
+			};
+		},
+		[ clientId ]
+	);
+
+	// Styles
+	const textStyle = usePreferredColorSchemeStyle(
+		styles.sheet__text,
+		styles[ 'sheet__text--dark' ]
+	);
+	const titleStyle = usePreferredColorSchemeStyle(
+		styles.sheet__title,
+		styles[ 'sheet__title--dark' ]
+	);
+	const descriptionStyle = usePreferredColorSchemeStyle(
+		styles.sheet__description,
+		styles[ 'sheet__description--dark' ]
+	);
+	const iconStyle = usePreferredColorSchemeStyle(
+		styles.sheet__icon,
+		styles[ 'sheet__icon--dark' ]
+	);
+	const actionButtonStyle = usePreferredColorSchemeStyle(
+		styles[ 'sheet__action-button' ],
+		styles[ 'sheet__action-button--dark' ]
+	);
+
+	const blockTitle =
+		customBlockTitle || getBlockType( blockName )?.title || blockName;
+
+	const requestFallback = useCallback( () => {
+		if (
+			canEnableUnsupportedBlockEditor &&
+			isUnsupportedBlockEditorSupported === false
+		) {
+			onCloseSheet();
+			setSendButtonPressMessage( true );
+		} else {
+			onCloseSheet();
+			setSendFallbackMessage( true );
+		}
+	}, [
+		canEnableUnsupportedBlockEditor,
+		isUnsupportedBlockEditorSupported,
+		onCloseSheet,
+	] );
+
+	const canUseWebEditor =
+		( isUnsupportedBlockEditorSupported ||
+			canEnableUnsupportedBlockEditor ) &&
+		isEditableInUnsupportedBlockEditor;
+	const actions = [
+		...[
+			canUseWebEditor && {
+				label: actionButtonLabel || __( 'Edit using web editor' ),
+				onPress: requestFallback,
+			},
+		],
+		...customActions,
+	].filter( Boolean );
```

Comment:
I noticed in the prototype builds that sites that do not have access to the Unsupported Block Editor display an empty action. 

| iOS | Android |
| - | - |
| ![empty-action-ios](https://github.com/WordPress/gutenberg/assets/438664/850cd043-d0fb-47aa-8ec6-44801f374e09) | ![empty-action-android](https://github.com/WordPress/gutenberg/assets/438664/746fffcf-9b20-4f4f-aa11-5daa557c2ae2) |

Reply:
Ah, that's interesting. I bumped into this issue while testing before setting the PR as ready to review, but I thought I had fixed it 🤔. I'll take a look in case I overlooked anything in the changes. Thanks!

Reply:
Maybe the prototype builds are merely using an outdated Gutenberg ref? I did not check.

Reply:
Ah, that's likely the case as I haven't recently updated the host app PRs. I'll update them asap.